src/main/webapp/js/common/react/components/DataFetcher.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * @fileoverview This component is used to fetch data for another component.
+ * Using this component means that you do not need to create your own fetch
+ * boilerplate.
+ *
+ * To use this component, you need two things:
+ * 1) A function that returns a fetch request for the data needed. This will be
+ * provided to the DataFetcher through its props. See {@link DataFetcherProps}.
+ * One important thing to note is that the DataFetcher supports abort signalling,
+ * which is used to cancel the fetch request in the case that the component is
+ * removed before the fetch can be completed. See this for how to use the abort
+ * signal in your fetch request:
+ * https://stackoverflow.com/questions/31061838/how-do-i-cancel-an-http-fetch-request
```

Comment:
```suggestion
 * removed before the fetch can be completed.
```
Info is already present below.